hw/xbox/xid-sdl.c

```diff
@@ -511,18 +564,11 @@ static void usb_xbox_gamepad_realize(USBDevice *dev, Error **errp)
     SDL_JoystickEventState(SDL_ENABLE);
 #endif
 
-#ifdef FORCE_FEEDBACK
-    s->sdl_haptic = SDL_HapticOpenFromJoystick(s->sdl_gamepad);
-    if (s->sdl_haptic == NULL) {
-        fprintf(stderr, "Joystick doesn't support haptic\n");
-    } else {
-        if ((SDL_HapticQuery(s->sdl_haptic) & SDL_HAPTIC_LEFTRIGHT) == 0) {
-            fprintf(stderr, "Joystick doesn't support necessary haptic effect\n");
-            SDL_HapticClose(s->sdl_haptic);
-            s->sdl_haptic = NULL;
-        }
+    if (SDL_InitSubSystem(SDL_INIT_HAPTIC)) {
+        fprintf(stderr, "SDL failed to initialize haptic feedback subsystem\n");
     }
-#endif
+
+    haptic_init(s);
```

Comment:
If `SDL_InitSubSystem(SDL_INIT_HAPTIC)` fails, we still do `haptic_init` anyway?

Reply:
We should probably not run `haptic_init` after the init fails. Technically the first call (`SDL_HapticOpened`) will always fail without the subsystem init, but skipping over this line is a good idea.

I return the error handling from `haptic_init` but I don't think handling anything at this level is really that useful, if something goes wrong inside `haptic_init` we log enough to `stderr` that a user should understand what happened. I return the error values just as a standard practice if we ever want to build some extra logic when an error occurs in the future.

If it is misleading though, I can just rip it out have have `haptic_init` return void.